### PR TITLE
Atgrid update

### PR DESCRIPTION
**Pull request by tczorro**
Update multiple atomic grid code review problems include default values, function names etc.

### CODE REVIEW

**Comment by FarnazH** [comment]
Thanks @tczorro. There are changes made to ` src/grid/rtransform.py` in this PR that doesn't belong here, I think, as it was part of #104.

**Comment by tczorro** [comment]
@FarnazH Those are just changes from the previous PR. I didn't start from a new branch.

File: src/grid/atomgrid.py [hunk 1]
```diff
@@ -165,7 +169,7 @@ def special_init(
         # 0.5rad <= r < rad, angular grid with degree 7
         # rad <= r < 1.5rad, angular grid with degree 5
         # 1.5rad <= r, angular grid with degree 3
-        >>> atgrid = AtomicGrid.special_init(rgrid, radius, degs, r_sectors, center)
+        >>> atgrid = AtomGrid.special_init(rgrid, radius, degs, r_sectors, center)
```
Comment (FarnazH) [inline src/grid/atomgrid.py:172]: Change `special_init` to `from_pruned`.

File: src/grid/atomgrid.py [hunk 2]
```diff
@@ -180,31 +184,37 @@ def special_init(
...
+            if center is None
+            else np.asarray(center, dtype=float)
+        )
         cls._input_type_check(rgrid, center)
         degs = cls._generate_degree_from_radius(rgrid, radius, r_sectors, degs)
         return cls(rgrid, degs=degs, center=center, rotate=rotate)
 
     @property
-    def rad_grid(self):
+    def rgrid(self):  # name rgrid
```
Comment (FarnazH) [inline src/grid/atomgrid.py:217]: Remove `# name rgrid` comment.

File: src/grid/atomgrid.py [hunk 3]
```diff
@@ -234,7 +244,7 @@ def l_max(self):
         """int: Largest angular degree L value in angular grids."""
         return np.max(self._rad_degs)
 
-    def get_shell_grid(self, index, r_sq=True):
+    def get_shell_grid(self, index, r_sq=True):  # reason
```
Comment (FarnazH) [inline src/grid/atomgrid.py:247]: I think this was a question for you to add the reason for defining this function no? In any event, the `# reason` comment should be removed.

File: src/grid/atomgrid.py [hunk 4]
```diff
@@ -261,44 +271,38 @@ def get_shell_grid(self, index, r_sq=True):
             new_wts = wts
         return AngularGrid(pts, new_wts)
 
-    def convert_point_to_sph(self, points):
-        """Convert a set of Cartesian points to sphercial coordinates.
+    def convert_cart_to_sph(self, points=None, center=None):
+        """Convert a set of points from cartesian to spherical coordinates.
+
+        The spherical center is selected as the atom if not given
+        The points is selected as all the atomic grid points if not given
```
Comment (FarnazH) [inline src/grid/atomgrid.py:278]: Isn't it better to add these to the arguments definition in the `Parameters` section? It better reflects what is used if they are `None`.

File: src/grid/lebedev.py [hunk 5]
```diff
@@ -196,32 +199,19 @@ def _select_grid_type(*, degree=None, size=None):
         )
 
 
-def _load_grid_filename(degree: int, size: int):
-    """Construct Lebedev file name for given degree and size.
+def _load_grid_filename(degree: int, size: int):  # TODO: why i put it.
```
Comment (FarnazH) [inline src/grid/lebedev.py:202]: Please remove TODO comment.

File: src/grid/lebedev.py [hunk 6]
```diff
@@ -117,12 +118,12 @@ def generate_lebedev_grid(*, degree=None, size=None):
...
     """
     degree, size = _select_grid_type(degree=degree, size=size)
-    points, weights = _load_grid_arrays(_load_grid_filename(degree, size))
+    points, weights = _load_grid_filename(degree, size)
     # set weights to 4\pi
     return AngularGrid(points, weights * 4 * np.pi)
 
 
-def match_degree(degree_nums):
+def match_degree(degree_nums):  # clear way coding
```
Comment (FarnazH) [inline src/grid/lebedev.py:126]: If this is not addressed in this PR, please change `# clear way coding` to `# TODO: clear way coding`, so it is not easier to track and address later.

File: src/grid/atomgrid.py [hunk 7]
```diff
@@ -459,14 +427,18 @@ def _generate_atomic_grid(rad_grid, degs, rotate=False):
                 new_points = sphere_grid.points @ rot_mt
                 sphere_grid = AngularGrid(new_points, sphere_grid.weights)
             # construct atomic grid with each radial point and each spherical shell
-            points, weights = AtomicGrid._generate_sphere_grid(rad_grid[i], sphere_grid)
+            rad_order = 2
+            # compute points
+            points = sphere_grid.points * rgrid[i].points
+            # compute weights
+            weights = (
+                sphere_grid.weights * rgrid[i].weights * rgrid[i].points ** rad_order
```
Comment (FarnazH) [inline src/grid/atomgrid.py:435]: Is there any reason to define `rad_order = 2`? I think you can just replace it on line 435.

**Pull request closed without merging**